index.bs

```diff
@@ -290,41 +283,50 @@ spec: HEADER-STRUCTURE; urlPrefix: https://httpwg.org/http-extensions/draft-ietf
-        <ol>
-          <li>If <var>item</var> is [=same origin-domain=] with
-          <var>origin</var>, then return true.</li>
-        </ol>
-      </li>
-      <li>return false.</li>
-    </ol>
+    <div algorithm="matches">
+      <p>To determine whether an <a>allowlist</a> <dfn>matches</dfn> an origin
+      <var>origin</var>, run these steps:
+
+      1. If the <a>allowlist</a> is <a>the special value <code>*</code></a>,
+         then return true.
+
+      1. Otherwise, [=set/for each=] |item| in the <a>allowlist</a>:
+
+        1. If |item| is [=same origin-domain=] with <var>origin</var>, then
+           return true.
+
+      1. Return false.
+    </div>
   </section>
   <section>
     <h3 id="default-allowlists">Default Allowlists</h3>
     <p>Every <a>policy-controlled feature</a> has a <dfn export
     lt="default allowlist|default allowlists"
     for="policy-controlled feature">default allowlist</dfn>. The <a>default
-    allowlist</a> determines whether the feature is allowed in a document with
-    no declared policy in a [=/top-level traversable=], and also whether access
-    to the feature is automatically delegated to documents in [=child
-    navigables=].</p>
+    allowlist</a> determines whether the feature is allowed in a {{Document}} with
+    no <a for="permissions policy">declared policy</a> in a [=/top-level
+    traversable=], and also whether access to the feature is automatically
+    delegated to documents in [=child navigables=].</p>
     <p>The <a>default allowlist</a> for a <a
     data-lt="policy-controlled feature">feature</a> is one of these values:</p>
     <dl>
       <dt><dfn for="default allowlist" export><code>*</code></dfn></dt>
-      <dd>The feature is allowed in documents in top-level traversables by
-      default, and when allowed, is allowed by default to documents in child
-      navigables.</dd>
+      <dd>The feature is allowed in {{Document}}s in [=/top-level
+      traversables=] by default, as well as those in all [=child navigables=].
+      It can be disallowed in [=child navigables=] by explicitly supplying a
+      [=container policy=] on the [=navigable container=] that overrides this
+      default (or in any [=/navigable=], by delivering the {{Document}} with a
+      suitable <a http-header>`Permissions-Policy`</a> header).</dd>
       <dt><dfn for="default allowlist" export><code>'self'</code></dfn></dt>
-      <dd>The feature is allowed in documents in top-level traversables by
-      default, and when allowed, is allowed by default to same-origin domain
-      documents in child navigables, but is disallowed by default in
-      cross-origin documents in child navigables.</dd>
+      <dd>The feature is allowed in [=navigable/active document|documents=] in
+      [=/top-level traversables=] by default, as well as those in [=child
+      navigables=] whose [=navigable/active document|document=] is
+      [=same origin=] with the [=/top-level traversable=]'s [=navigable/active
```

Comment:
I think this should still read something like "`...is [=same origin=] with its [=navigable/parent=]'s [=navigable/active document|document=], when allowed in that {{Document}}`"

Reply:
Aren't both true? For example, my understanding is that in `A(B(A))` if the feature is disabled in `B`, the inner `A` still has it enabled if the outer `A` does, since the inner `A` could just synchronously script its grandparent and use the feature anyways (so there's no need to turn it off for the inner-most `A` here).

I think your statement is also true as well, but maybe we could outline both scenarios here?

Reply:
In `A(B(A))`, if it's disabled in `B`, it will be disabled in inner `A` as well -- it's true that inner `A` could usually call back to outer `A` to use a feature (depending on the feature; some would act differently if called in different frames), but to make the algorithms consistent, we disable it -- generally keeping to "if a feature is disabled in any frame, then subframes cannot re-enable it".

So there's no same-origin-with-top-level check (or really a same-origin-with-any-other-frame-with-the-feature-enabled check); we just check whether each frame embedding creates an origin boundary, and determine inheritance at that point.

Reply:
Got it. I've clarified this here.